libs/components/text-editor/src/lib/modules/text-editor/text-editor.component.html

```diff
@@ -3,12 +3,20 @@
   [ngClass]="{
     'sky-text-editor-disabled': disabled,
     'sky-text-editor-wrapper-focused': editorFocused,
-    'sky-text-editor-invalid': ngControl.errors,
+    'sky-text-editor-invalid':
+      ngControl.errors && (ngControl.touched || ngControl.dirty),
     'sky-margin-stacked-xs': !!hintText
   }"
 >
   <div class="sky-text-editor-label-wrapper">
-    <label *ngIf="labelText" class="sky-control-label">{{ labelText }}</label>
+    <label
+      *ngIf="labelText"
+      class="sky-control-label"
+      [ngClass]="{
+        'sky-control-label-required': requiredState.isRequired()
```

Comment:
We've historically tried to not bind to function calls. This is however how we are getting away with not having a `requiredChange` event. Thoughts? When we did all the strict mode stuff - we were pretty against doing functional calls on bindings like this due to the fact that the function will be called on every change detection cycle.

Reply:
We do [something similar](https://github.com/blackbaud/skyux/blob/main/libs/components/forms/src/lib/modules/input-box/input-box.component.ts#L209-L229) with the input box's required and disabled attributes. While we could add a `requiredChange` event, it would only fire when the attribute changes; we'd have to add a `statusChanges` subscription in the directive for the abstract control required state as well. Since there's no clear "required change" event from Angular, and the way to listen for that event isn't linear, I think it's okay to make this an exception to the rule.